src/OpenTelemetry/Internal/TagTransformer.cs

```diff
@@ -18,6 +18,8 @@ namespace OpenTelemetry.Internal;
 
 internal abstract class TagTransformer<T>
 {
+    public static Action<string, string> LogUnsupportedAttributeType = null;
```

Comment:
I feel this is becoming bit complicated - e.g. if some code accidentally update this part, it might cause undefined behavior? How do we prevent other OpenTelemetry components from updating this global state?

Reply:
The components that use `TagTransformer` would have their own instances of its implementation. There is no global state here.

For example, if you were to use `ConsoleExporter` and `OtlpExporter` together, each of those exporters would be operating on their instance of `TagTransformer` as they link this file into their respective dlls.

Reply:
Ideally, we should not use the Sdk project to host these utility files which are only used by other components and not the Sdk itself. We could work on moving such files out of the Sdk project as a follow-up PR.